### PR TITLE
Add README for document_repository module

### PR DESCRIPTION
**Pull request by kongtiaowang**
This pull request adds README for document_repository module.

### CODE REVIEW

File: modules/document_repository/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Document Repository
+
+## Purpose
+
+The Document Repository is a useful tool that provides studies with a centralized location for important documents. Files may be uploaded and organized under any user-defined category. Subcategories can also be defined by the user to be nested under parent categories.
```
Comment (driusan) [inline modules/document_repository/README.md]: I think it should be up to the users to decide if it's useful, I'd take out the adjective. It can also be used for unimportant documents. Didn't we decide a while ago that the document repository should be for non-candidate documents? (Candidate documents go in the media module.)

The second sentence should be in the scope, not the purpose, and I'm not sure that the third is required.

File: modules/document_repository/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Intended Users
+ 
+ All Loirs user could view, update, edit and delete the upload files with right permissions.
```
Comment (driusan) [inline modules/document_repository/README.md]: LORIS, not Loirs. I think it's mostly used by data entry staff, isn't it?

File: modules/document_repository/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Scope
+
+ For all of the Loris users.
```
Comment (driusan) [inline modules/document_repository/README.md]: I think the scope should say that it's for study documents, not candidate documents (which go into the media module)

File: modules/document_repository/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## Permissions
+
+ A user has "document_repository_view" permission could view and upload files in Document Repository.
```
Comment (driusan) [inline modules/document_repository/README.md]: missing the word "that" before "has" and "the" after ("A user that has the [..]") and "can view", not "could view" (same comments apply to the delete permission below)

File: modules/document_repository/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ A user has "document_repository_delete" permission could delete files in Document Repository.
+
+## Configurations
+
+
+- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions. Give full permissions to store documents on server:
+
+chmod 777 /var/www/loris/modules/document_repository/user_uploads
+Note that the path is assumed to be var/www/loris however your own path may be var/www/<project-name>, depending on your setup.
```
Comment (driusan) [inline modules/document_repository/README.md]: I think this should just say that the web server needs to be able to write to it and not try to give the command, it's up to the sysadmins to be able to configure the permissions. and it should just refer to the `$LORIS/modules/document_repository/user_uploads` directory.

File: modules/document_repository/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+- A mail server will be required for sending out email notifications about Document Repository updates. In addition, in the Configuration Module under the "WWW" settings section, "Main LORIS URL" must be set. Also see more information on configuring Apache's file upload size limit.
+
+- The table "document_repository_categories" in the database can be used to initialize the name of the category.
```
Comment (driusan) [inline modules/document_repository/README.md]: I'm not sure what you mean by "initialize".. isn't this table just populated by the doc repo front end?

File: modules/document_repository/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+## Configurations
+
+
+- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions. Give full permissions to store documents on server:
+
+chmod 777 /var/www/loris/modules/document_repository/user_uploads
+Note that the path is assumed to be var/www/loris however your own path may be var/www/<project-name>, depending on your setup.
+
+- A mail server will be required for sending out email notifications about Document Repository updates. In addition, in the Configuration Module under the "WWW" settings section, "Main LORIS URL" must be set. Also see more information on configuring Apache's file upload size limit.
```
Comment (driusan) [inline modules/document_repository/README.md]: This should specify which php.ini settings affect it (I think they're the same as the media module or imaging uploader?) The other READMEs refer to the configurations as they're specified in the code, not the front end.

I think "In addition" just needs to be a different bullet point.

File: modules/document_repository/README.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ The Document Repository is a tool that provides studies with a centralized location for non-candidate documents. Files may be uploaded and organized under any user-defined category.
+
+## Intended Users
+ 
+ All Loris user and data entry staff could view, update, edit and delete the upload files with right permissions.
+
+## Scope
+
+ It's for all non-candidate documents.
+
```
Comment (christinerogers) [inline modules/document_repository/README.md]: I think when you associate a Site to a document, only users from that Site can see it.   (If so, please definitely note this functionality in the spec.)

File: modules/document_repository/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+```
+session.gc_maxlifetime = 10800  // After this number of seconds, stored data will be seen as 'garbage' and cleaned up by the garbage collection process.
+max_input_time = 10800          // Maximum amount of time each script may spend parsing request data (in seconds)
+max_execution_time = 10800      // Maximum execution time of each script (in seconds)
+upload_max_filesize = 1024M     // Maximum allowed size for uploaded files.
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
+
+- A mail server will be required for sending out email notifications about Document Repository updates. 
+
```
Comment (christinerogers) [inline modules/document_repository/README.md]: The documents are uploaded and stored under the modules/document_repository/user_uploads/directory.   Note this directory should be (made?) writable, and care should also be taken to back up these documents / not clobber them when updating code or migrating machines.

Interactions: 
Document repository (notifications?) are visible on the dashboard.

File: modules/document_repository/README.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
+# Document Repository
+
+## Purpose
+
+ The Document Repository is a tool that provides studies with a centralized location for non-candidate documents. Files may be uploaded and organized under any user-defined category.
+
+## Intended Users
+ 
+ All Loris user and data entry staff could view, update, edit and delete the upload files with right permissions.
```
Comment (christinerogers) [inline modules/document_repository/README.md]: I would edit to say: 
"Loris users can view, download and upload files, edit information about files, and delete files given appropriate permissions.
This module is used primarily by study coordinators/administrators who share study protocols and manuals, and by data entry staff who consult these documents. "

File: modules/document_repository/README.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
+# Document Repository
+
+## Purpose
+
+ The Document Repository is a tool that provides studies with a centralized location for non-candidate documents. Files may be uploaded and organized under any user-defined category.
```
Comment (christinerogers) [inline modules/document_repository/README.md]: * Change: "non-candidate documents" to "documents relevant to the study across multiple candidates" 
* "user-defined category"  --> please add note below under Configurations describing how users can add a Category.

File: modules/document_repository/README.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+## Intended Users
+ 
+ All Loris user and data entry staff could view, update, edit and delete the upload files with right permissions.
+
+## Scope
+
+ It's for all non-candidate documents.
```
Comment (christinerogers) [inline modules/document_repository/README.md]: Change: "non-candidate documents" to "documents relevant to the study across multiple candidates  (as opposed to Media module)"

**Comment by kongtiaowang** [comment]
@christinerogers 
Thanks for the reviews, I have modified the PR. Please review again.

File: modules/document_repository/README.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
+# Document Repository
+
+## Purpose
+
+ The Document Repository is a tool that provides studies with a centralized location for non-candidate documents relevant to the study across multiple candidates.
+ Files may be uploaded and organized under any user-defined category.
+
+## Intended Users
+
+ Loris users can view, download and upload files, edit information about files, and delete files given appropriate permissions.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Would write LORIS all upper case instead of Loris.
Also, instead of "download and upload files" would tend to write "download/upload files"

File: modules/document_repository/README.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
+# Document Repository
+
+## Purpose
+
+ The Document Repository is a tool that provides studies with a centralized location for non-candidate documents relevant to the study across multiple candidates.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Would simplify the first sentence a bit:
"The Document Repository is a tool that provides a centralized location for non-candidate documents relevant to the study. For example, copies of non-filled instruments, publications, manual of operations..."

File: modules/document_repository/README.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+ A user that has "document_repository_view" permission can view and upload files in Document Repository.
+ A user that has "document_repository_delete" permission can delete files in Document Repository.
+
+## Configurations
+
+
+- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions. Give full permissions to store documents on server.
+
+- Enable uploads files:please update the php.ini apache configuration file.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Should probably be "To enable file's upload: please update the php.ini apache configuration file."

Comment (cmadjar) [reply modules/document_repository/README.md]: missing a space after ":"

File: modules/document_repository/README.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+- Enable uploads files:please update the php.ini apache configuration file. 
+```
+session.gc_maxlifetime = 10800  // After this number of seconds, stored data will be seen as 'garbage' and cleaned up by the garbage collection process.
+max_input_time = 10800          // Maximum amount of time each script may spend parsing request data (in seconds)
+max_execution_time = 10800      // Maximum execution time of each script (in seconds)
+upload_max_filesize = 1024M     // Maximum allowed size for uploaded files.
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
+
+- Enable uploads folder:
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Should probably be: "To enable folder upload:"

File: modules/document_repository/README.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+"chmod 777 /var/www/loris/modules/document_repository/user_uploads".
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Should be in code format

File: modules/document_repository/README.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+Note that the path is assumed to be var/www/loris however your own path may be var/www/<project-name>, depending on your setup.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: paths should be in code format. Is there a reason there is not leading / in front of the paths?

File: modules/document_repository/README.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+
+- A mail server will be required for sending out email notifications about Document Repository updates. 
+
+- Interactions: Document repository notifications are visible on the dashboard panel.
+  
+- Create user-defined category in the "document_repository_categories" table. For example "INSERT INTO `document_repository_categories` VALUES (1,'text',0,NULL),(2,'photo',0,NULL),(3,'pdf',0,NULL);".
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Table name and SQL command should be in code format

File: modules/document_repository/README.md [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+upload_max_filesize = 1024M     // Maximum allowed size for uploaded files.
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
+
+- To enable folder upload:
+```
+chmod 777 /var/www/loris/modules/document_repository/user_uploads
+
+```
+Note that the path is assumed to be /var/www/loris however your own path may be /var/www/<project-name>, depending on your setup.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Sorry, I missed this in my first series of comment. The path should probably be in code format, no? That would fix the issue that we don't see `<project-name>` in the display of the MD file.
Same for `php.ini` in line 26.

File: modules/document_repository/README.md [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+
+- A mail server will be required for sending out email notifications about Document Repository updates. 
+
+- Interactions: Document repository notifications are visible on the dashboard panel.
+  
+- Create user-defined category in the "document_repository_categories" table. For example
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Sorry, I am being very picky. There should be ":" after "For example"

File: modules/document_repository/README.md (hunk 20 above)
Comment (cmadjar) [reply modules/document_repository/README.md]: Would also have a comma after the `/var/www/loris` path so that it looks like this:

_Note that the path is assumed to be `/var/www/loris`, however your own path may be `/var/www/<project-name>`, depending on your setup._

This "note" sentence should probably be between brackets or in italic so that it is clear that the 3 following items belong to the Configuration section and are not items belonging to the note (I got confused for a little while when reading it)

File: modules/document_repository/README.md [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+max_input_time = 10800          // Maximum amount of time each script may spend parsing request data (in seconds)
+max_execution_time = 10800      // Maximum execution time of each script (in seconds)
+upload_max_filesize = 1024M     // Maximum allowed size for uploaded files.
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
+
+- To enable folder upload:
+```
+chmod 777 /var/www/loris/modules/document_repository/user_uploads
+
```
Comment (cmadjar) [inline modules/document_repository/README.md]: remove the extra empty line in the `chmod` code

**Comment by kongtiaowang** [comment]
@cmadjar Thanks, you are right. I will make it more clearly.

File: modules/document_repository/README.md [hunk 23]
```diff
@@ -0,0 +1,46 @@
+# Document Repository
+
+## Purpose
+
+The Document Repository is a tool that provides a centralized location for non-candidate documents relevant to the study. For example, copies of non-filled instruments, publications, manual of operations.
```
Comment (zaliqarosli) [inline modules/document_repository/README.md]: suggestion: "The Document Repository is a tool that provides a centralized location for non-candidate documents that are relevant to the study i.e. copies of unfilled instruments, publications, and manual of operations."

File: modules/document_repository/README.md [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+
+## Intended Users
+
+ LORIS users can view, download/upload files, edit information about files, and delete files given appropriate permissions.
```
Comment (zaliqarosli) [inline modules/document_repository/README.md]: suggestion: "Given the appropriate permissions, LORIS users can view, download, upload, and delete files, as well as edit information pertaining to the files."

File: modules/document_repository/README.md [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+This module is used primarily by study coordinators/administrators who share study protocols and manuals, and by data entry staff who consult these documents.
```
Comment (zaliqarosli) [inline modules/document_repository/README.md]: suggestion: "This module is used primarily by study coordinators/administrators who share study protocols and manuals, as well as data entry staff who consult these documents."

File: modules/document_repository/README.md (hunk 23 above)
Comment (driusan) [reply modules/document_repository/README.md]: I think "unfilled versions of instruments" would be better than "copies of unfilled instruments"

Comment (driusan) [reply modules/document_repository/README.md]: and maybe "and/or" instead of "and" (or even just "or")

File: modules/document_repository/README.md [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+ The document can only be seen by Users from the Site that the document is associated with.
+
+## Permissions
+
+ A user that has "document_repository_view" permission can view and upload files in Document Repository.
+ A user that has "document_repository_delete" permission can delete files in Document Repository.
+
+## Configurations
+
+- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions. Give full permissions to store documents on server.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: I would tend to put the following sentence in the Scope section:
"The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions."

And I would only focus on the configuration part in this section.

Also, I would write: "To give full permissions to store documents on the server:"

File: modules/document_repository/README.md [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
+
+- To enable folder upload:
+```
+chmod 777 /var/www/loris/modules/document_repository/user_uploads
+```
+[Note that the path is assumed to be ```/var/www/loris```, however your own path may be ```/var/www/<project-name>``` , depending on your setup.]
+
+- A mail server will be required for sending out email notifications about Document Repository updates.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Would rephrase to "A mail server is required to send out email notification regarding the Document Repository updates."

File: modules/document_repository/README.md [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+
+- To enable folder upload:
+```
+chmod 777 /var/www/loris/modules/document_repository/user_uploads
+```
+[Note that the path is assumed to be ```/var/www/loris```, however your own path may be ```/var/www/<project-name>``` , depending on your setup.]
+
+- A mail server will be required for sending out email notifications about Document Repository updates. 
+
+- Interactions: Document repository notifications are visible on the dashboard panel.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: Would rephrase a bit to: "Notifications from the Document Repository module are visible on the Dashboard panel."

File: modules/document_repository/README.md [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+  
+- Create user-defined category in the "document_repository_categories" table. For example:
```
Comment (cmadjar) [inline modules/document_repository/README.md]: I don't understand what you mean by this last one. Could you clarify?

File: modules/document_repository/README.md [hunk 30]
```diff
@@ -0,0 +1,41 @@
...
+ The document can only be seen by Users from the Site that the document is associated with.
+
+## Permissions
+
+ A user that has "document_repository_view" permission can view and upload files in Document Repository.
+ A user that has "document_repository_delete" permission can delete files in Document Repository.
+
+## Configurations
+
+- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions.Give full permissions to store documents on server. To give full permissions to store documents on the server:
```
Comment (cmadjar) [inline modules/document_repository/README.md]: I would split the first sentence in two:
"The Document Repository enables users to upload and organize project files of any type. These files can easily be viewable for users with appropriate permissions."

Also, remove duplicated sentence in **_bold-italic_** below:
"permissions.**_Give full permissions to store documents on server._** To give"

File: modules/document_repository/README.md [hunk 31]
```diff
@@ -0,0 +1,41 @@
+# Document Repository
+
+## Purpose
+
+ The Document Repository is a tool that provides a centralized location for non-candidate documents that are relevant to the study (i.e. unfilled versions of instruments, publications, or manual of operations).
+
+## Intended Users
+
+ Given the appropriate permissions, LORIS users can view, download, upload, and delete files, as well as edit information pertaining to those files.
+ This module is used primarily by study coordinators/administrators who share study protocols and manuals, as well as data entry staff who consult these documents.
```
Comment (cmadjar) [inline modules/document_repository/README.md]: I would not limit access to the doc repository to data entry staff. People doing analyses might need to have access to the doc repo to have access to publications and be able to see what the instruments look like or the project documentation. I think the doc repository should be accessible to all users as it contains pertinent information regarding the project description.

File: modules/document_repository/README.md [hunk 32]
```diff
@@ -20,7 +20,8 @@
 
 ## Configurations
 
-- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions.Give full permissions to store documents on server. To give full permissions to store documents on the server: 
+- The Document Repository enables users to upload and organize project files of any type that can easily be viewable for users with appropriate permissions.
```
Comment (zaliqarosli) [inline modules/document_repository/README.md]: Could we elaborate on what "of any type" means?

Also .. "that can easily be viewed by users with the appropriate permissions."

File: modules/document_repository/README.md [hunk 33]
```diff
@@ -0,0 +1,42 @@
...
+ The Document Repository is a tool that provides a centralized location for non-candidate documents that are relevant to the study (i.e. unfilled versions of instruments, publications, or manual of operations).
+
+## Intended Users
+
+ LORIS users can view, download, upload, and delete files, as well as edit information pertaining to those files.
+ This module is used primarily by study coordinators/administrators who share study protocols and manuals, as well as data entry staff who consult these documents.
+
+## Scope
+
+ Any LORIS user can edit and delete the all files in the Document Repository module.
```
Comment (zaliqarosli) [inline modules/document_repository/README.md]: Typo: "Any LORIS user can edit and delete ALL THE files in the Document Repository module"

File: modules/document_repository/README.md [hunk 34]
```diff
@@ -0,0 +1,38 @@
...
+
+## Permissions
+
+ A user that has "document_repository_view" permission can view and upload files in Document Repository.
+ A user that has "document_repository_delete" permission can delete files in Document Repository.
+
+## Configurations
+
+- The Document Repository enables users to upload and organize project files of any type (PDF,Photo,Txt...) that can easily be viewable for users with appropriate permissions ("document_repository_view").
+- Give full permissions to store documents on server:
```
Comment (driusan) [inline modules/document_repository/README.md]: This sentence is out of place. I'm not sure what it's saying.

File: modules/document_repository/README.md [hunk 35]
```diff
@@ -0,0 +1,38 @@
...
+- Give full permissions to store documents on server: 
+```
+session.gc_maxlifetime = 10800  // After this number of seconds, stored data will be seen as 'garbage' and cleaned up by the garbage collection process.
+max_input_time = 10800          // Maximum amount of time each script may spend parsing request data (in seconds)
+max_execution_time = 10800      // Maximum execution time of each script (in seconds)
+upload_max_filesize = 1024M     // Maximum allowed size for uploaded files.
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
+
+[Note that the path is assumed to be ```/var/www/loris```, however your own path may be ```/var/www/<project-name>``` , depending on your setup.]
```
Comment (driusan) [inline modules/document_repository/README.md]: There's no reference to `/var/www/loris` above.

File: modules/document_repository/README.md [hunk 36]
```diff
@@ -0,0 +1,38 @@
...
+
+- The Document Repository enables users to upload and organize project files of any type (PDF,Photo,Txt...) that can easily be viewable for users with appropriate permissions ("document_repository_view").
+- Give full permissions to store documents on server: 
+```
+session.gc_maxlifetime = 10800  // After this number of seconds, stored data will be seen as 'garbage' and cleaned up by the garbage collection process.
+max_input_time = 10800          // Maximum amount of time each script may spend parsing request data (in seconds)
+max_execution_time = 10800      // Maximum execution time of each script (in seconds)
+upload_max_filesize = 1024M     // Maximum allowed size for uploaded files.
+post_max_size = 1024M           // Maximum size of POST data that PHP will accept.
+```
```
Comment (driusan) [inline modules/document_repository/README.md]: There should be a sentence giving this context and saying where these settings go.

Comment (kongtiaowang) [reply modules/document_repository/README.md]: done.
